clang-tools-extra/clangd/FindSymbols.cpp

```diff
@@ -391,6 +392,16 @@ class DocumentOutline {
         D = TD;
     }
 
+    // Friend declarations should be traversed if and only if
+    // they are also defined here.
```

Comment:
we aren't really visiting friend decl, we're visiting the underlying function definition.

can you also leave a comment mentioning why we special case this? (e.g. `FriendDecls don't act as `DeclContext`s but they might wrap a function definition that won't be visible through other means in the AST. Hence unwrap it here instead.`)